cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -750,5 +750,309 @@ TEST(Substrait, ExtensionSetFromPlanMissingFunc) {
           &ext_set));
 }
 
+TEST(Substrait, JoinPlanBasic) {
```

Comment:
These plans are very verbose.  I can't think of any good trick to shorten it but I worry a little about how much raw substrait json we are going to have in all our unit tests and that becoming a maintenance burden.  Any ideas?

Reply:
Yes, I completely agree with you. One thing we can do is, putting them in a Submodule like the datasets and use it when it is required. But I am not sure how good that option would be. It will again add the same burden of keeping that in the memory (too big), but avoid us keeping the text in the Arrow repo itself. 

Is this feasible?

Reply:
I'm not so much worried about where the text lives as I am worried about maintaining the text itself.  On the other hand, any approach I can think of for automatically generating the text devolves into creating Ibis/dplyr for C++ :laughing: 

Still, those tools exist for a reason, writing these things manually is cumbersome.

Reply:
Actually... @saulpw might be able to help us here with his JSOM stuff.  I'll ping him.

Reply:
@westonpace can we generate it, may be we can incoporate `isthmus` to generate it during the tests and we only write a SQL query. I don't know if that is feasible as there are slight differences between the things we digest right now. Just a thought 😀

Reply:
Sounds good. Looking forward for this :)